lib/java/desktop-environment/src/test/java/org/enso/desktopenvironment/TrashTest.java

```diff
@@ -0,0 +1,82 @@
+package org.enso.desktopenvironment;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.commons.io.FileUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TrashTest {
+
+  private static final Trash trash = Platform.getTrash();
+
+  @Test
+  public void isSupported() {
+    Assert.assertTrue(trash.isSupported());
+  }
+
+  @Test
+  public void moveToTrashFile() throws IOException {
+    var path = writeTempFile("moveToTrashFile");
+
+    try {
+      Assert.assertTrue(trash.moveToTrash(path));
+      Assert.assertFalse(trash.moveToTrash(path));
+    } finally {
+      FileUtils.deleteQuietly(path.toFile());
```

Comment:
You can use `@Rule TemporaryFolder` just like in https://github.com/enso-org/enso/blob/develop/engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/exports/ExportExtensionMethodTest.java#L23 and avoid the need to delete the path yourself - JUnit runner will do that for you.